### PR TITLE
feat: add support to context in navigation

### PR DESCRIPTION
**Pull request by uziasferreirazup**
### Related Issues

https://github.com/ZupIT/beagle/issues/1740

### Description and Example

Internal project inside the company needs native support in the Beagle Core so that you can pass parameters from one screen to another without using the GlobalContext. This feature is also useful for the Beagle as a whole, even without internal projects. That's why it's important to implement it in the core, as open-source.

### Checklist

Please, check if these important points are met using `[x]`:

- [X] I read the [PR Guide] and followed the process outlined there for submitting this PR.
- [X] I avoided _breaking changes_ by not changing public APIs that people rely on. 
- [X] I am willing to follow up on review comments in a timely manner.
- [X] I have made the documentation changes or I created an issue explaining how to document this change on [Docs issues](https://github.com/ZupIT/beagle-docs/issues). Please link the issue here:

[PR Guide]: https://github.com/ZupIT/beagle-android/blob/main/doc/contributing/pull_requests.md

### CODE REVIEW

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 1]
```diff
@@ -17,19 +17,29 @@
 package br.com.zup.beagle.android.view
 
 import android.os.Bundle
+import android.os.Handler
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:20]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L20"></span>The import 'android.os.Handler' is unused.</td>
    </tr>
  </tbody>
</table>

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 2]
```diff
@@ -17,19 +17,29 @@
...
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
 import android.widget.FrameLayout
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.Lifecycle
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:26]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L26"></span>The import 'androidx.lifecycle.Lifecycle' is unused.</td>
    </tr>
  </tbody>
</table>

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 3]
```diff
@@ -17,19 +17,29 @@
...
+import androidx.lifecycle.LifecycleObserver
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:27]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L27"></span>The import 'androidx.lifecycle.LifecycleObserver' is unused.</td>
    </tr>
  </tbody>
</table>

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 4]
```diff
@@ -17,19 +17,29 @@
...
+import androidx.lifecycle.OnLifecycleEvent
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:28]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L28"></span>The import 'androidx.lifecycle.OnLifecycleEvent' is unused.</td>
    </tr>
  </tbody>
</table>

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 5]
```diff
@@ -17,19 +17,29 @@
...
+import br.com.zup.beagle.android.context.ContextData
 import br.com.zup.beagle.android.data.serializer.BeagleSerializer
 import br.com.zup.beagle.android.utils.applyBackgroundFromWindowBackgroundTheme
 import br.com.zup.beagle.android.utils.toView
 import br.com.zup.beagle.android.view.viewmodel.AnalyticsViewModel
 import br.com.zup.beagle.android.view.viewmodel.BeagleScreenViewModel
+import br.com.zup.beagle.android.view.viewmodel.ScreenContextViewModel
 import br.com.zup.beagle.android.widget.UndefinedWidget
 import br.com.zup.beagle.android.widget.core.ServerDrivenComponent
+import android.os.Looper
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:39]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L39"></span>The import 'android.os.Looper' is unused.</td>
    </tr>
  </tbody>
</table>

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 6]
```diff
@@ -17,19 +17,29 @@
...
+import android.util.Log
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:40]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L40"></span>The import 'android.util.Log' is unused.</td>
    </tr>
  </tbody>
</table>

File: beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt [hunk 7]
```diff
@@ -17,19 +17,29 @@
...
+import java.util.logging.Logger
```
Comment (botbeaglezup) [inline beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt:41]: <table data-meta="generated_by_android">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleFragment.kt#L41"></span>The import 'java.util.logging.Logger' is unused.</td>
    </tr>
  </tbody>
</table>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          2 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/context/ContextDataManager.kt#L35">beagle/src/main/kotlin/br/com/zup/beagle/android/context/ContextDataManager.kt#L35</a> - Class 'ContextDataManager' with '13' functions detected. Defined threshold inside classes is set to '13'</td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/ZupIT/beagle-android/blob/e6e212aeddcd8b91f8c031a9c9d31a35779e51a8/beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleActivity.kt#L33">beagle/src/main/kotlin/br/com/zup/beagle/android/view/BeagleActivity.kt#L33</a> - The import 'androidx.lifecycle.Observer' is unused.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Comment by botbeaglezup** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>

### Tests: 

Name | Classname | Time|
--- | --- | ---|
Should call Analytics Service with correct parameters | br.com.zup.beagle.android.view.viewmodel.AnalyticsViewModelTest$CreateActionReport | 0.058|


### Tests: 

Name | Classname | Time|
--- | --- | ---|
Should call Analytics Service with correct parameters | br.com.zup.beagle.android.view.viewmodel.AnalyticsViewModelTest$CreateScreenReport | 0.007|


<p align="right" data-meta="generated_by_android">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

**Review by matheusribeirozup — APPROVED** [review]

**Pull request merged**